src/eth.rs

```diff
@@ -613,6 +613,13 @@ impl<P> EspEth<P> {
         Ok(eth)
     }
 
+    pub fn with_handle<F, T>(&self, f: F) -> T
```

Comment:
Semantically this should be `&mut self`, not `&self` if it can be used to modify the ethmac